packages/@expo/cli/src/start/server/type-generation/routes.ts

```diff
@@ -34,7 +35,37 @@ export interface SetupTypedRoutesOptions {
   routerDirectory: string;
 }
 
-export async function setupTypedRoutes({
+export async function setupTypedRoutes(options: SetupTypedRoutesOptions) {
+  const typedRoutesModule = resolveFrom.silent(
+    options.projectRoot,
+    'expo-router/build/typed-routes'
+  );
+  return typedRoutesModule ? typedRoutes(typedRoutesModule, options) : legacyTypedRoutes(options);
+}
+
+async function typedRoutes(
+  typedRoutesModulePath: any,
+  { server, metro, typesDirectory, projectRoot, routerDirectory }: SetupTypedRoutesOptions
+) {
+  process.env.EXPO_ROUTER_APP_ROOT = routerDirectory;
+
+  const typedRoutesModule = require(typedRoutesModulePath);
+
+  if (metro && server) {
```

Comment:
Should probably be an assertion

Reply:
I think I prefer adding a comment explaining why these values could be false.